tests/units/anta_tests/__init__.py

```diff
@@ -3,10 +3,43 @@
 # that can be found in the LICENSE file.
 """Tests for anta.tests module."""
 
+from __future__ import annotations
+
 import asyncio
-from typing import Any
+from typing import TYPE_CHECKING, Any, Literal, TypedDict
+
+if TYPE_CHECKING:
+    from typing_extensions import NotRequired  # NOTE: required to support Python < 3.11 https://peps.python.org/pep-0655/#usage-in-python-3-11
+
+    from anta.device import AntaDevice
+    from anta.models import AntaTest
+
+
+class AtomicResult(TypedDict):
+    """Expected atomic result of a unit test of an AntaTest subclass."""
+
+    result: Literal["success", "failure", "skipped"]  # TODO: Refactor tests and use AntaTestStatus
+    description: str
+    messages: NotRequired[list[str]]
+    inputs: NotRequired[dict[str, Any]]
+
 
-from anta.device import AntaDevice
+class Expected(TypedDict):
+    """Expected result of a unit test of an AntaTest subclass."""
+
+    result: Literal["success", "failure", "skipped"]  # TODO: Refactor tests and use AntaTestStatus
+    messages: NotRequired[list[str]]
+    atomic_results: NotRequired[list[AtomicResult]]
+
+
+class AntaUnitTest(TypedDict):
+    """The parameters required for a unit test of an AntaTest subclass."""
+
+    name: str  # TODO: Refactor tests and change the DATA constant type as dictionary instead of list[AntaUnitTest] to avoid test duplicates.
```

Comment:
same - do we have lots of duplicates?

Reply:
This has been cleaned as part of the benchmark PR, and there is a some check there to be removed: https://github.com/aristanetworks/anta/blob/main/tests/benchmark/test_anta.py#L82

https://github.com/aristanetworks/anta/issues/996